### PR TITLE
Documentation for field_masking_span query

### LINKED ISSUE #20293: Documentation for field_masking_span query
It looks like [`field_masking_span` query](https://github.com/elastic/elasticsearch/issues/471) has been supported for a long time but [remains undocumented](https://www.elastic.co/guide/en/elasticsearch/reference/master/span-queries.html). 

It should be included in the documentation for the latest 1.x, 2.x and 5.x releases.

### PR DESCRIPTION
**Pull request by yevhen**
Closes #20293

### CODE REVIEW

File: docs/reference/query-dsl/span-field-masking-query.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+This can be used to support queries like `span-near` or `span-or` across different fields, which is not ordinarily permitted.
+
+Span field masking query is invaluable in conjunction with *multi-fields* when same content is indexed with multiple analyzers. For instance we could index a field with the standard analyzer which breaks text up into words, and again with the english analyzer which stems words into their root form. That will add more precision to queries.
+
+Example:
+
+[source,js]
+--------------------------------------------------
+GET /_search
```
Comment (clintongormley) [inline docs/reference/query-dsl/span-field-masking-query.asciidoc]: There's a typo in the example, please use the one below.  Also `collect_payloads` is deprecated, so I've removed it from my version.

```
GET /_search
{
  "query": {
    "span_near": {
      "clauses": [
        {
          "span_term": {
            "text": "quick brown"
          }
        },
        {
          "field_masking_span": {
            "query": {
              "span_term": {
                "text.stems": "fox"
              }
            },
            "field": "text"
          }
        }
      ],
      "slop": 5,
      "in_order": false
    }
  }
}
```

File: docs/reference/query-dsl/span-field-masking-query.asciidoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[[query-dsl-span-field-masking-query]]
+=== Span Field Masking Query
+
+Wrapper to allow span queries participate in composite single-field span queries by 'lying' about their search field. The span field masking query maps to Lucene `SpanFieldMaskingQuery`
```
Comment (clintongormley) [inline docs/reference/query-dsl/span-field-masking-query.asciidoc]: TO participate

Comment (clintongormley) [inline docs/reference/query-dsl/span-field-masking-query.asciidoc]: to Lucene -> to Lucene's

File: docs/reference/query-dsl/span-field-masking-query.asciidoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+This can be used to support queries like `span-near` or `span-or` across different fields, which is not ordinarily permitted.
+
+Span field masking query is invaluable in conjunction with *multi-fields* when same content is indexed with multiple analyzers. For instance we could index a field with the standard analyzer which breaks text up into words, and again with the english analyzer which stems words into their root form. That will add more precision to queries.
+
```
Comment (clintongormley) [inline docs/reference/query-dsl/span-field-masking-query.asciidoc]: I'd remove "That will add more precision to queries" as it is pretty vague, and I can see users following the advice without understanding the reasoning.

**Comment by clintongormley** [comment]
thanks for the PR @yevhen 

I've made a few comments - please could you fix them and push again?

thanks

**Comment by yevhen** [comment]
@clintongormley done

**Comment by clintongormley** [comment]
looks good - thanks @yevhen , merged

**Pull request merged**